packages/plugin-ext-vscode/src/browser/plugin-vscode-commands-contribution.ts

```diff
@@ -54,5 +68,30 @@ export class PluginVscodeCommandsContribution implements CommandContribution {
                 this.contextKeyService.createKey(String(contextKey), contextValue);
             }
         });
+
+        commands.registerCommand(VscodeCommands.PREVIEW_HTML, {
+            isVisible: () => true,
+            // tslint:disable-next-line: no-any
+            execute: (resource: URI, position?: any, label?: string, options?: any) => {
+                label = label || resource.fsPath;
+                const view = new WebviewWidget(label, {allowScripts: true}, {});
+                this.resorces(new TheiaURI(resource)).then(res => {
```

Comment:
could you please use async/await to be easier to read ?

Reply:
looks like no, it because here https://github.com/theia-ide/theia/blob/master/packages/core/src/common/resource.ts#L93 code not `async`

Reply:
```typescript
commands.registerCommand(VscodeCommands.PREVIEW_HTML, {
            isVisible: () => true,
            // tslint:disable-next-line: no-any
            execute: async (resource: URI, position?: any, label?: string, options?: any) => {
                label = label || resource.fsPath;
                const view = new WebviewWidget(label, { allowScripts: true }, {});
                const res = await this.resorces(new TheiaURI(resource));
                const str = await res.readContents();
                const html = this.getHtml(str);
                this.shell.addWidget(view, { area: 'main', mode: 'split-right' });
                this.shell.activateWidget(view.id);
                view.setHTML(html);

                const editorWidget = await this.editorManager.getOrCreateByUri(new TheiaURI(resource));
                editorWidget.editor.onDocumentContentChanged(listener => {
                    view.setHTML(this.getHtml(editorWidget.editor.document.getText()));
                });

            }
        }
        );
```

Reply:
WOW! Thanks, you open new feature for me!